Source/WebCore/editing/MarkupAccumulator.cpp

```diff
@@ -217,18 +211,48 @@ void MarkupAccumulator::serializeNodesWithNamespaces(Node& targetNode, Serialize
         namespaceHash.set(XMLNames::xmlNamespaceURI->impl(), xmlAtom().impl());
     }
 
-    if (root == SerializedNodes::SubtreeIncludingNode)
-        startAppendingNode(targetNode, &namespaceHash);
+    const Node* current = &targetNode;
+    do {
+        bool shouldAppendNode = !(current == &targetNode && root != SerializedNodes::SubtreeIncludingNode);
+        if (shouldAppendNode)
+            startAppendingNode(*current, &namespaceHash);
```

Comment:
This seems a behavior change compared to previous implementation: we used to not append the target node if its tag name is skipped, now we append it but not append its children. Is this intended?

Reply:
Oh yeah, that's a good catch!

Reply:
Ah, you're right, thanks!